cpp/v2/test/api_test_v2.cc

```diff
@@ -0,0 +1,262 @@
+#include <api_v2.h>
+#include <gtest/gtest.h>
+
+#include <iostream>
+
+using namespace ray;
+
+void hello() { std::cout << "hello\n"; }
+
+void dummy() { std::cout << "dummy\n"; }
+
+int bar(int a) {
+  std::cout << "bar"
+            << ", arg: " << a << "\n";
+  return a;
+}
+
+namespace my_namespace {
+int bar(int a) {
+  std::cout << "bar in namespace"
+            << ", arg: " << a << "\n";
+  return a + 1;
+}
+}  // namespace my_namespace
+
+int add(int a, int b) { return a + b; }
+
+void not_registered_func() {}
+
+int overload_func() {
+  std::cout << "overload_func no arguments\n";
+  return 0;
+}
+int overload_func(int i) {
+  std::cout << "overload_func one argument\n";
+  return i;
+}
+int overload_func(int i, int j) {
+  std::cout << "overload_func two arguments\n";
+  return i + j;
+}
+
+struct Base {
+  Base() = default;
+  Base(int) {}
+  Base(int, ObjectRef<int>) {}
```

Comment:
why do we need `ObjectRef<int>`? shouldn't `Base(int, int)` be able to take in `Base(int, ObjectRef<int>)`

Reply:
You could use Base(int, int) or Base(int, ObjectRef<int>), it's up to the user, ray api supports both.